frontend/src/metabase/visualizations/lib/renderer_utils.js

```diff
@@ -161,7 +161,9 @@ export function getDatas({ settings, series }, warn) {
 
     return rows.map(row => {
       const [x, ...rest] = row;
-      const newRow = [parseXValue(x, parseOptions, warn), ...rest];
+      const { unit } = parseOptions;
+      const xValue = parseXValue(x, parseOptions, warn);
+      const newRow = [xValue.startOf(unit), ...rest];
```

Comment:
By using `xValue.startOf(unit)`, the date will be reseted to the beginning of the `unit` (month in the examples) and key strings will be retrieved correctly in the data map with filled missing values:
-  with `type/Instant` for `2016-01-01T01:00:00+01:00` => `2015-12-31T23:00:00.000Z`
-  with `type/DateTime` for `2019-01-01T00:00:00+01:00` => `2018-12-31T23:00:00.000Z`